src/geo/transform.js

```diff
@@ -1262,6 +1271,20 @@ class Transform {
         return posMatrix;
     }
 
+    calculateDistanceTileMatrix(unwrappedTileID: UnwrappedTileID): Float32Array {
+        const fogTileMatrixKey = unwrappedTileID.key;
+        const cache = this._distanceMatrixCache;
+        if (cache[fogTileMatrixKey]) {
+            return cache[fogTileMatrixKey];
+        }
+
+        const posMatrix = this.calculatePosMatrix(unwrappedTileID, this.cameraWorldSize);
+        mat4.multiply(posMatrix, this.centerDistanceMatrix, posMatrix);
+
+        cache[fogTileMatrixKey] = new Float32Array(posMatrix);
```

Comment:
It's been on my todo list for a while to switch these all to plain Arrays so that they'd be double precision with GL also still accepting them.